src/planner/planner.c

```diff
@@ -482,21 +482,21 @@ timescaledb_planner(Query *parse, int cursor_opts, ParamListInfo bound_params)
 			{
 				reset_fetcher_type = true;
 
-				if (ts_guc_remote_data_fetcher == AutoFetcherType)
+				if (context.num_distributed_tables >= 2)
```

Comment:
The idea was for manual `ts_guc_remote_data_fetcher` to have preference over everything else. So if you set it manually to row-by-row when it can't work, it is expected to fail. I think this is more straightforward to use, i.e. it doesn't quietly ignore the explicit setting in some cases.

The default is `auto`, so why does it fail in your case, did you change it to row-by-row?

Reply:
The error message should be less cryptic though

Reply:
Hmm yes this is due to the guc setting, similar planner gucs in postgres will fall back to working configuration to prevent running in a dead-end. e.g. disable_seqscan will still do seq scan if nothing else works. I think this behaviour is preferable over the current behaviour. If we encounter a situation that is known to not work we should not produce weird errors but instead fall back to working configuration.

Reply:
Changed it to throw error instead